awx/ui_next/src/screens/ObservabilityMetrics/LineChart.jsx

```diff
@@ -0,0 +1,272 @@
+      .append('text')
+      .attr('class', 'legend-title')
+      .attr('x', '100')
+      .attr('y', '50')
+      .text('Legend');
+
+    legendContainer.data(data, (d, i) => {
+      if (d?.name) {
+        const legendItemContainer = legendContainer
+          .append('div')
+          .style('display', 'flex')
+          .attr('id', 'legend-item-container')
+          .style('padding-left', '20px');
+
+        legendItemContainer
+          .append('div')
+          .style('background-color', color(i))
+          .style('height', '8px')
+          .style('width', '8px')
+          .style('border-radius', '50%')
+          .style('padding', '5px')
+          .style('margin-top', '6px');
+
+        legendItemContainer
+          .append('text')
+          .style('padding-left', '20px')
+          .text(d.name);
+      }
+    });
+
+    // Add help text to top of chart
+
+    d3.select('#chart')
+      .append('div')
+      .attr('class', 'help-text')
+      .style('padding-left', '50px')
+      .style('padding-top', '20px')
+      .text(helpText);
+
+    const svg = d3
+      .select('#chart')
+      .append('svg')
+      .attr('width', `${width + margin}px`)
+      .attr('height', `${height + margin}px`)
+      .append('g')
+      .attr('transform', `translate(${margin}, ${margin})`);
+
+    /* Add line into SVG */
+    const line = d3
+      .line()
+      .curve(d3.curveMonotoneX)
+      .x(d => xScale(d.x))
+      .y(d => yScale(d.y));
+
+    const lines = svg.append('g');
+
+    lines
+      .selectAll('.line-group')
+      .data(data)
+      .enter()
+      .append('g')
+      .attr('class', 'line-group')
+      .append('path')
+      .attr('class', 'line')
+      .style('fill', 'none')
+      .attr('d', d => line(d.values))
+      .style('stroke', (d, i) => color(i))
+      .style('stroke-width', '3px');
+
+    /* Add circles in the line */
+    lines
+      .selectAll('circle-group')
+      .data(data)
+      .enter()
+      .append('g')
+      .style('fill', (d, i) => color(i))
+      .selectAll('circle')
+      .data(d => d.values)
+      .enter()
+      .append('g')
+      .attr('class', 'circle')
+      .on('mouseover', (d, i) => {
+        if (data.length) {
+          renderTooltip(d, i);
+        }
+      })
+      .on('mouseout', () => {
+        removeTooltip();
+      })
+      .append('circle')
+      .attr('cx', d => xScale(d.x))
+      .attr('cy', d => yScale(d.y))
+      .attr('r', circleRadius)
+      .on('mouseover', () => {
+        d3.select(this)
+          .transition()
+          .duration(duration)
+          .attr('r', circleRadiusHover);
+      })
+      .on('mouseout', () => {
+        d3.select(this)
+          .transition()
+          .duration(duration)
+          .attr('r', circleRadius);
+      });
+
+    /* Add Axis into SVG */
+    const xAxis = d3
+      .axisBottom(xScale)
+      .ticks(data[0].values.length > 5 ? data[0].values.length : 5);
+    const yAxis = d3.axisLeft(yScale).ticks(5);
+
```

Comment:
![Screenshot from 2021-04-06 09-29-03](https://user-images.githubusercontent.com/9753817/113721659-b71c4280-96bd-11eb-8f99-bdb9aee7f0a3.png)

 The "values" and "counts" labels are confusing to me. "Counts" isn't really accurate for the horizontal axis and, if anything, it's more suitable for the vertical axis because they're typically showing counts of event types received. I think we should either remove the axis labels or change "counts" to "time" and "values" to "counts".

Reply:
Yeah I struggled with the names here since, especially the x-axis, isn't really anything of value.  I'm on board with removing them.